kong/router.lua

```diff
@@ -1236,13 +1257,29 @@ function _M.new(routes)
 
     req_method = upper(req_method)
 
-    if req_host ~= "" then
-      -- strip port number if given because matching ignores ports
-      local idx = find(req_host, ":", 2, true)
-      if idx then
-        ctx.req_host = sub(req_host, 1, idx - 1)
+    -- req_host might have port or maybe not, host_no_port definitely doesn't
+    -- if there wasn't a port, req_port is assumed to be the default port
+    -- according the protocol scheme
+    local host_with_port = req_host
+    local host_no_port, req_port = utils.split_port(host_with_port)
+    if not req_port then
+      req_port = 80
+      if req_scheme == 'https' then
+        req_port = 443
+      end
+
+      -- a literal IPv6 address needs the format [host]:port,
+      -- the fastest way to check is if the first byte was stripped by split_port()
+      -- otherwise, check for a colon
```

Comment:
Wouldn't it be a bit faster (and perhaps a bit more explicit) to do check for the square bracket character directly, instead of "checking if it has changed"?

```
if byte(host_with_port) == LEFTBRACKET or ...
```

Of course disregard this if you end up moving `host_with_port` inside `utils.split_port`

Reply:
disregarded